Libraries/libcar/Headers/car/Writer.h

```diff
@@ -60,6 +67,17 @@ class Writer {
      */
     void addRendition(Rendition const &rendition);
 
+    /*
+     * Add a rendition for a facet, optimized for fast editing of CAR files
+     */
+    void addRendition(void *key, size_t key_len, void *value, size_t value_len);
+
+    /*
+     * The key format, optional and determined automatically if omitted.
+     */
+    ext::optional<struct car_key_format *> &keyfmt()
+    { return _keyfmt; }
```

Comment:
That's true, but if you're specifying an optional value, then you probably have a reason.

The best way to handle this I think is to catch this error in the serialize functions.